db/columns.py

```diff
@@ -166,15 +183,21 @@ def create_column(engine, table_oid, column_data):
     if sa_type is None:
         logger.warning("Requested type not supported. falling back to VARCHAR")
         sa_type = supported_types["VARCHAR"]
+        column_type_options = {}
     table = tables.reflect_table_from_oid(table_oid, engine)
     column = MathesarColumn(
-        column_data[NAME], sa_type, nullable=column_nullable,
+        column_data[NAME], sa_type(**column_type_options), nullable=column_nullable,
     )
     with engine.begin() as conn:
         ctx = MigrationContext.configure(conn)
         op = Operations(ctx)
         op.add_column(table.name, column, schema=table.schema)
-    return tables.reflect_table_from_oid(table_oid, engine).columns[column_data[NAME]]
+
+    new_column = MathesarColumn.from_column(
```

Comment:
Could we use `get_mathesar_column_with_engine` here?

Reply:
Yep, good eye.